third_party/amd/backend/driver.py

```diff
@@ -1,482 +1,152 @@
+
+import torch
+
 from triton.backends.compiler import GPUTarget
 from triton.backends.driver import GPUDriver
+from triton.backends.amd import hip
 
-dirname = os.path.dirname(os.path.realpath(__file__))
-include_dir = [os.path.join(dirname, "include")]
 
+def check(status):
+    if status != 0:
+        raise RuntimeError(f"HIP Error {status}, {ctypes.string_at(hip.hipGetErrorString(status)).decode()}")
 
-def _find_already_mmapped_dylib_on_linux(lib_name):
-    import platform
-    if platform.system() != 'Linux':
-        return None
 
-    # Use dl_iterate_phdr to walk through the list of shared libraries at runtime.
-    # See https://www.man7.org/linux/man-pages/man3/dl_iterate_phdr.3.html for details.
+def get_pointer(a):
+    if a is None:
+        return None
+    attributes = hip.hipPointerAttribute_t()
+    data_ptr = hip.hipDeviceptr_t(a.data_ptr())
+    check(hip.hipPointerGetAttributes(ctypes.byref(attributes), data_ptr))
+    return hip.hipDeviceptr_t(attributes.devicePointer)
+
+
+def ty_to_ctype(ty: str):
+    if ty.startswith("*"):
+        return hip.hipDeviceptr_t
+    elif ty == "i1":
+        return ctypes.c_int32
+    elif ty == "i8":
+        return ctypes.c_int8
+    elif ty == "i16":
+        return ctypes.c_int16
+    elif ty == "i32":
+        return ctypes.c_int32
+    elif ty == "i64":
+        return ctypes.c_int64
+    elif ty == "u1":
+        return ctypes.c_uint32
+    elif ty == "u8":
+        return ctypes.c_uint8
+    elif ty == "u16":
+        return ctypes.c_uint16
+    elif ty == "u32":
+        return ctypes.c_uint32
+    elif ty == "u64":
+        return ctypes.c_uint64
+    elif ty == "fp32":
+        return ctypes.c_float
+    elif ty == "fp64":
+        return ctypes.c_double
+    # see https://github.com/llvm/llvm-project/blob/main/mlir/python/mlir/runtime/np_to_memref.py#L17-L43
+    elif ty == "fp16":
+        return ctypes.c_int16
+    elif ty == "bf16":
+        return ctypes.c_int16
+    else:
+        raise NotImplementedError(f"{ty=} not supported")
+
+
+class HIPLauncher:
 
-    import ctypes
-    from ctypes import c_char, c_int, c_size_t, c_void_p, c_char_p, POINTER
+    def __init__(self, src, metadata):
+        constants = src.constants if hasattr(src, "constants") else dict()
+        cst_key = lambda i: src.fn.arg_names.index(i) if isinstance(i, str) else i
+        constants = {cst_key(key): value for key, value in constants.items()}
+        signature = {cst_key(key): value for key, value in src.signature.items()}
+        from triton.runtime.jit import TensorWrapper
+
+        def launch(*args, **kwargs):
+            assert not kwargs
+            gridX, gridY, gridZ, stream, function, kernel_metadata, launch_metadata, launch_enter_hook, launch_exit_hook, *args = args
+            num_warps, num_ctas, shared_memory, clusterDimX, clusterDimY, clusterDimZ = kernel_metadata
+            if launch_enter_hook or launch_exit_hook:
+                raise NotImplementedError
```

Comment:
Presumably these are actually used somehow? But I couldn't tell from the currently emitted trampoline. Happy to add support for this as well.

Reply:
These hooks are used by proton to record profile scope information.